File: FichaRpg.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>
#include <locale.h>

/*
    Criação de um programa em C, para criação de uma ficha de personagem para um RPG
*/
/*
    Descrição: procedimento que recebe a escolha de classe e monta a ficha apartir daquela classe
    Procedimento do codigo: A partir  da escolha do usuario, ira ter condições onde a ficha será montada para cada classe
*/
void preencherfichaManu(int classe, char *nome, char *raca, int *dados, int *armor, int *vida, int *nivel, int *xp) {
  // Declarando variáveis
  int i;
  char Classe[50];
  char atributos[6][15] = {"Forca",     "Destreza",     "Constituicao",  "Sabedoria", "Inteligencia", "Carisma"};
  
  
  printf("Digite o nome do seu personagem: ");
  scanf("%s", nome);
  printf("Escolha sua raça que deseja entre essas:\nHumano\nOrc\nElfo\n");
  scanf("%s", raca);

  // 1 - classe guerreiro
  if (classe == 1) {
    strcpy(Classe, "Guerriro");
    for (i = 0; i < 6; i++) {
      printf("Digite o %d dado do atributo %s: ", i + 1, atributos[i]);
      scanf("%d", &dados[i]);
    }
    printf("\nArmor: ");
    scanf("%d", &armor);
    fflush(stdin);
    printf("\nVida: ");
    scanf("%d", &vida);
    fflush(stdin);
    printf("\nNivel: ");
    scanf("%d", &nivel);
    fflush(stdin);
    printf("\nXP: ");
    scanf("%d", &xp);
    fflush(stdin);
    //Caso o raça sejá Humana
    if (raca == 'Humana') {
      for (i = 0; i < 6; i++) {
        dados[i] = dados[i] + 1;
      }

    }
    //Caso a raça escolhida sejá Orc
    else if (raca == 'Orc') {
      dados[1] = dados[1] + 2;
      dados[3] = dados[3] + 1;
      
    }
    //Caso a raça seja Elfica
    else if (raca == 'Elfo') {
      dados[2] = dados[2] + 1;
    }
  }
  //2-Condição para a segunda classe Mago
  if (classe == 2) {
    strcpy(Classe, "Mago");
    for (i = 0; i < 6; i++) {
      printf("Digite o %d dado do atributo %s: ", i + 1, atributos[i]);
      scanf("%d", &dados[i]);
    }
    printf("\nArmor: ");
    scanf("%d", &armor);
    fflush(stdin);
    printf("\nVida: ");
    scanf("%d", &vida);
    fflush(stdin);
    printf("\nNivel: ");
    scanf("%d", &nivel);
    fflush(stdin);
    printf("\nXP: ");
    scanf("%d", &xp);
    fflush(stdin);
    
    if (raca == 'Humana') {
      for (i = 0; i < 6; i++) {
        dados[i] = dados[i] + 1;
      }

    } else if (raca == 'Orc') {
      dados[1] = dados[1] + 2;
      dados[3] = dados[3] + 1;
    } else if (raca == 'Elfo') {
      dados[2] = dados[2] + 1;
    }
  }
  //3-classe druida
  if (classe == 3) {
    strcpy(Classe, "Druida");
    for (i = 0; i < 6; i++) {
      printf("Digite o %d dado do atributo %s: ", i + 1, atributos[i]);
      scanf("%d", &dados[i]);
    }
    printf("\nArmor: ");
    scanf("%d", &armor);
    fflush(stdin);
    printf("\nVida: ");
    scanf("%d", &vida);
    fflush(stdin);
    printf("\nNivel: ");
    scanf("%d", &nivel);
    fflush(stdin);
    printf("\nXP: ");
    scanf("%d", &xp);
    fflush(stdin);
    //Caso o raça sejá Humana
    if (strcmp(raca, "Humano") == 0) {
      for (i = 0; i < 6; i++) {
        dados[i] = dados[i] + 1;
      }
    }
    else if (strcmp(raca, "Orc") == 0) {
      dados[1] = dados[1] + 2;
      dados[3] = dados[3] + 1;
    }
    else if (strcmp(raca, "Elfo") == 0) {
      dados[2] = dados[2] + 1;
    }
  }
  
}
/*
    Descrição:
        procedimento que monta ficha do personagem automatica rolando os dados, sendo eles:
        -4d6 para atributos
        
*/
void preencherficharaut(int classe, char *nome, char *raca, int *dados, int *armor, int *vida, int *nivel, int *xp) {
    int i, j, soma_dados = 0;
    char Classe[50];
    char atributos[6][15] = {"Forca", "Destreza", "Constituicao", "Sabedoria", "Inteligencia", "Carisma"};
  
    srand(time(NULL));
  
    printf("Digite o nome do seu personagem: ");
    scanf("%s", nome);
  
    printf("Escolha sua raça entre essas:\nHumano\nOrc\nElfo\n");
    scanf("%s", raca);

    // 1 - classe guerreiro
    if (classe == 1) {
        strcpy(Classe, "Guerreiro");
        for (i = 0; i < 6; i++) {
            soma_dados = 0;
            for (j = 0; j < 4; j++) {
                // Gerar um número aleatório entre 1 e 6
                int numero = rand() % 6 + 1;
                soma_dados += numero;
            }
            dados[i] = soma_dados;
        }

        printf("\nArmor: ");
        scanf("%d", armor);
        fflush(stdin);
        printf("\nVida: ");
        scanf("%d", vida);
        fflush(stdin);
        printf("\nNivel: ");
        scanf("%d", nivel);
        fflush(stdin);
        printf("\nXP: ");
        scanf("%d", &xp);
        fflush(stdin);
      
        // Caso a raça seja Humana
        if (strcmp(raca, "Humano") == 0) {
            for (i = 0; i < 6; i++) {
                dados[i] = dados[i] + 1;
            }
        }
        else if (strcmp(raca, "Orc") == 0) {
            dados[1] = dados[1] + 2;
            dados[3] = dados[3] + 1;
        }
        else if (strcmp(raca, "Elfo") == 0) {
            dados[2] = dados[2] + 1;
        }
    }
    // 2 - classe Mago
    else if (classe == 2) {
        strcpy(Classe, "Mago");
        for (i = 0; i < 6; i++) {
            soma_dados = 0;
            for (j = 0; j < 4; j++) {
                // Gerar um número aleatório entre 1 e 6
                int numero = rand() % 6 + 1;
                soma_dados += numero;
            }
            dados[i] = soma_dados;
        }
      
        printf("\nArmor: ");
        scanf("%d", armor);
        fflush(stdin);
        printf("\nVida: ");
        scanf("%d", vida);
        fflush(stdin);
        printf("\nNivel: ");
        scanf("%d", nivel);
        fflush(stdin);
        printf("\nXP: ");
        scanf("%d", &xp);
        fflush(stdin);
      
        // Caso a raça seja Humana
        if (strcmp(raca, "Humano") == 0) {
            for (i = 0; i < 6; i++) {
                dados[i] = dados[i] + 1;
            }
        }
        else if (strcmp(raca, "Orc") == 0) {
            dados[1] = dados[1] + 2;
            dados[3] = dados[3] + 1;
        }
        else if (strcmp(raca, "Elfo") == 0) {
            dados[2] = dados[2] + 1;
        }
    }
    // 3 - classe druida
    else if (classe == 3) {
        strcpy(Classe, "Druida");
        for (i = 0; i < 6; i++) {
            soma_dados = 0;
            for (j = 0; j < 4; j++) {
                // Gerar um número aleatório entre 1 e 6
                int numero = rand() % 6 + 1;
                soma_dados += numero;
            }
            dados[i] = soma_dados;
        }
      
        printf("\nArmor: ");
        scanf("%d", armor);
        fflush(stdin);
        printf("\nVida: ");
        scanf("%d", vida);
        fflush(stdin);
        printf("\nNivel: ");
        scanf("%d", nivel);
        fflush(stdin);
        printf("\nXP: ");
        scanf("%d", &xp);
        fflush(stdin);
      
        // Caso a raça seja Humana
        if (strcmp(raca, "Humano") == 0) {
            for (i = 0; i < 6; i++) {
                dados[i] = dados[i] + 1;
            }
        }
        else if (strcmp(raca, "Orc") == 0) {
            dados[1] = dados[1] + 2;
            dados[3] = dados[3] + 1;
        }
        else if (strcmp(raca, "Elfo") == 0) {
            dados[2] = dados[2] + 1;
        }
    }
}

void imprimirFicha(int classe, char *nome, char *raca, int *dados, int *armor, int *vida, int *nivel, int *xp) {
    printf("\n--- FICHA DO PERSONAGEM ---\n");
    printf("Nome: %s\n", nome);
    printf("Classe: ");
    if (classe == 1)
        printf("Guerreiro\n");
    else if (classe == 2)
        printf("Mago\n");
    else if (classe == 3)
        printf("Druida\n");
    printf("Raça: %s\n", raca);
    printf("Atributos:\n");
    printf("Força: %d\n", dados[0]);
    printf("Destreza: %d\n", dados[1]);
    printf("Constituição: %d\n", dados[2]);
    printf("Sabedoria: %d\n", dados[3]);
    printf("Inteligência: %d\n", dados[4]);
    printf("Carisma: %d\n", dados[5]);
    printf("Armor: %d\n", *armor);
    printf("Vida: %d\n", *vida);
    printf("Nivel: %d\n", *nivel);
    printf("XP: %d\n", xp);
}


int main() {
    // Criação do menu, para o usuário escolher como quer sua ficha
    char nome[50];
    char raca[50];
    int armor=0;
    int vida=0;
    int nivel=0;
    int xp=0;
    int op = 0;
    int dados[6] = {0};  // Array to store attribute values
    int classe = 0;
    
    do {
        printf("\t-MONTE SUA FICHA DE RPG-\n");
        printf("Escolha uma das opcoes que desejar nos parametros 5.e:\n");
        printf("\t(1) Montar a ficha Manual\n");
        printf("\t(2) Montar a ficha Automática\n");
        printf("\t(3) Mostrar a ficha montada\n");
        printf("\t(0) Sair\n");
        
        scanf("%d", &op);
        
        switch (op) {
            case 1:
                printf("Escolha sua classe:\n(1) Guerreiro\n(2) Mago\n(3) Druida\n");
                scanf("%d", &classe);
                fflush(stdin);
                preencherfichaManu(classe, nome, raca, dados, &armor, &vida, &nivel, xp);
                break;
                
            case 2:
                printf("Escolha sua classe:\n(1) Guerreiro\n(2) Mago\n(3) Druida\n");
                scanf("%d", &classe);
                fflush(stdin);
                preencherficharaut(classe, nome, raca, dados, &armor, &vida, &nivel, &xp);
                break;
                
            case 3:                
                imprimirFicha(classe, nome, raca, dados, &armor, &vida, &nivel, &xp);
                break;
                
            case 0:
                printf("Saindo do programa...\n");
                break;
                
            default:
                printf("Opção inválida. Tente novamente.\n");
                break;
        }
        
    } while (op != 0);
    
    return 0;
}

File: FichaRPg.cs
using System;

class Program
{
    static void Main()
    {
        // Criação do menu, para o usuário escolher como quer sua ficha
        int classe=0;
        string nome = "";
        string raca = "";
        int[] dados = new int[6];
        int armor = 0;
        int vida = 0;
        int nivel = 0;
        int xp = 0;
        int op;

        do
        {
            Console.WriteLine("\t-MONTE SUA FICHA DE RPG-");
            Console.WriteLine("Escolha uma das opcoes que desejar nos parametros 5.e:");
            Console.WriteLine("\t(1) Montar a ficha Manual");
            Console.WriteLine("\t(2) Montar a ficha Automática");
            Console.WriteLine("\t(3) Mostrar a ficha montada");
            Console.WriteLine("\t(0) Sair");

            op = Convert.ToInt32(Console.ReadLine());

            switch (op)
            {
                case 1:
                    Console.WriteLine("Escolha sua classe:\n(1) Guerreiro\n(2) Mago\n(3) Druida");
                    classe = Convert.ToInt32(Console.ReadLine());
                    PreencherFichaManu(classe, dados);
                    break;

                case 2:
                    Console.WriteLine("Escolha sua classe:\n(1) Guerreiro\n(2) Mago\n(3) Druida");
                    classe = Convert.ToInt32(Console.ReadLine());
                    PreencherFichaRaut(classe, dados);
                    break;

                case 3:
                  
                    ImprimirFicha(classe, nome, raca, dados, armor, vida, nivel, xp);
                    break;

                case 0:
                    Console.WriteLine("Saindo do programa...");
                    break;

                default:
                    Console.WriteLine("Opção inválida. Tente novamente.");
                    break;
            }

        } while (op != 0);
    }
    //Criação da ficha manualmente
    static void PreencherFichaManu(int classe, int[] dados)
    {
        int vida = 0, armor = 0, nivel = 0, xp = 0, i;
        string nome;
        string raca;
        string[] atributos = { "Força", "Destreza", "Constituição", "Sabedoria", "Inteligência", "Carisma" };
        string Classe = " ";

        Console.WriteLine("Digite o nome do seu personagem: ");
        nome = Console.ReadLine();
        Console.WriteLine("Escolha sua raça que deseja entre essas:\nHumano\nOrc\nElfo");
        raca = Console.ReadLine();

        if (classe == 1)
        {
            Classe = "Guerreiro";
            for (i = 0; i < 6; i++)
            {
                Console.Write($"Digite o {i + 1}º dado do atributo {atributos[i]}: ");
                dados[i] = Convert.ToInt32(Console.ReadLine());
            }
            Console.Write("\nArmor: ");
            armor = Convert.ToInt32(Console.ReadLine());
            Console.Write("\nVida: ");
            vida = Convert.ToInt32(Console.ReadLine());
            Console.Write("\nNível: ");
            nivel = Convert.ToInt32(Console.ReadLine());
            Console.Write("\nXP: ");
            xp = Convert.ToInt32(Console.ReadLine());

            if (raca == "Humano")
            {
                for (i = 0; i < 6; i++)
                {
                    dados[i] = dados[i] + 1;
                }
            }
            else if (raca == "Orc")
            {
                dados[1] = dados[1] + 2;
                dados[3] = dados[3] + 1;
            }
            else if (raca == "Elfo")
            {
                dados[2] = dados[2] + 1;
            }
        }
        else if (classe == 2)
        {
            Classe = "Mago";
            for (i = 0; i < 6; i++)
            {
                Console.Write($"Digite o {i + 1}º dado do atributo {atributos[i]}: ");
                dados[i] = Convert.ToInt32(Console.ReadLine());
            }
            Console.Write("\nArmor: ");
            armor = Convert.ToInt32(Console.ReadLine());
            Console.Write("\nVida: ");
            vida = Convert.ToInt32(Console.ReadLine());
            Console.Write("\nNível: ");
            nivel = Convert.ToInt32(Console.ReadLine());
            Console.Write("\nXP: ");
            xp = Convert.ToInt32(Console.ReadLine());

            if (raca == "Humano")
            {
                for (i = 0; i < 6; i++)
                {
                    dados[i] = dados[i] + 1;
                }
            }
            else if (raca == "Orc")
            {
                dados[1] = dados[1] + 2;
                dados[3] = dados[3] + 1;
            }
            else if (raca == "Elfo")
            {
                dados[2] = dados[2] + 1;
            }
        }
        else if (classe == 3)
        {
            Classe = "Druida";
            for (i = 0; i < 6; i++)
            {
                Console.Write($"Digite o {i + 1}º dado do atributo {atributos[i]}: ");
                dados[i] = Convert.ToInt32(Console.ReadLine());
            }
            Console.Write("\nArmor: ");
            armor = Convert.ToInt32(Console.ReadLine());
            Console.Write("\nVida: ");
            vida = Convert.ToInt32(Console.ReadLine());
            Console.Write("\nNível: ");
            nivel = Convert.ToInt32(Console.ReadLine());
            Console.Write("\nXP: ");
            xp = Convert.ToInt32(Console.ReadLine());

            if (raca == "Humano")
            {
                for (i = 0; i < 6; i++)
                {
                    dados[i] = dados[i] + 1;
                }
            }
            else if (raca == "Orc")
            {
                dados[1] = dados[1] + 2;
                dados[3] = dados[3] + 1;
            }
            else if (raca == "Elfo")
            {
                dados[2] = dados[2] + 1;
            }
        }

        Console.WriteLine("Ficha preenchida.");
    }

    static void PreencherFichaRaut(int classe, int[] dados)
    {
       int vida = 0, armor = 0, nivel = 0, xp = 0, i;
        string nome;
        string raca;
        string[] atributos = { "Força", "Destreza", "Constituição", "Sabedoria", "Inteligência", "Carisma" };
        string Classe = "";

        Console.WriteLine("Digite o nome do seu personagem: ");
        nome = Console.ReadLine();
        Console.WriteLine("Escolha sua raça que deseja entre essas:\nHumano\nOrc\nElfo");
        raca = Console.ReadLine();
        Random random = new Random();

        if (classe == 1)
        {
            Classe = "Guerreiro";
            //Preenche rodando os dados
            for (i = 0; i < 6; i++){
                int soma_dados = 0;
                for (int j  = 0; j < 4; j++){
                    // Gerar um número aleatório entre 1 e 6
                    int numero = random.Next(1, 7);
                    soma_dados += numero;
                }
                dados[i] = soma_dados;
            }
            Console.Write("\nArmor: ");
            armor = Convert.ToInt32(Console.ReadLine());
            Console.Write("\nVida: ");
            vida = Convert.ToInt32(Console.ReadLine());
            Console.Write("\nNível: ");
            nivel = Convert.ToInt32(Console.ReadLine());
            Console.Write("\nXP: ");
            xp = Convert.ToInt32(Console.ReadLine());

            if (raca == "Humano")
            {
                for (i = 0; i < 6; i++)
                {
                    dados[i] = dados[i] + 1;
                }
            }
            else if (raca == "Orc")
            {
                dados[1] = dados[1] + 2;
                dados[3] = dados[3] + 1;
            }
            else if (raca == "Elfo")
            {
                dados[2] = dados[2] + 1;
            }
        }
        else if (classe == 2)
        {
            Classe = "Mago";
            //Preenche rodando os dados
            for (i = 0; i < 6; i++){
                int soma_dados = 0;
                for (int j = 0; j < 4; j++){
                    // Gerar um número aleatório entre 1 e 6
                    int numero = random.Next(1, 7);
                    soma_dados += numero;
                }
                dados[i] = soma_dados;
            }
            Console.Write("\nArmor: ");
            armor = Convert.ToInt32(Console.ReadLine());
            Console.Write("\nVida: ");
            vida = Convert.ToInt32(Console.ReadLine());
            Console.Write("\nNível: ");
            nivel = Convert.ToInt32(Console.ReadLine());
            Console.Write("\nXP: ");
            xp = Convert.ToInt32(Console.ReadLine());

            if (raca == "Humano")
            {
                for (i = 0; i < 6; i++)
                {
                    dados[i] = dados[i] + 1;
                }
            }
            else if (raca == "Orc")
            {
                dados[1] = dados[1] + 2;
                dados[3] = dados[3] + 1;
            }
            else if (raca == "Elfo")
            {
                dados[2] = dados[2] + 1;
            }
        }
        else if (classe == 3)
        {
            Classe = "Druida";
            //Preenche rodando os dados
            for (i = 0; i < 6; i++){
                int soma_dados = 0;
                for (int j = 0; j < 4; j++){
                    // Gerar um número aleatório entre 1 e 6
                    int numero = random.Next(1, 7);
                    soma_dados += numero;
                }
                dados[i] = soma_dados;
            }
            Console.Write("\nArmor: ");
            armor = Convert.ToInt32(Console.ReadLine());
            Console.Write("\nVida: ");
            vida = Convert.ToInt32(Console.ReadLine());
            Console.Write("\nNível: ");
            nivel = Convert.ToInt32(Console.ReadLine());
            Console.Write("\nXP: ");
            xp = Convert.ToInt32(Console.ReadLine());

            if (raca == "Humano")
            {
                for (i = 0; i < 6; i++)
                {
                    dados[i] = dados[i] + 1;
                }
            }
            else if (raca == "Orc")
            {
                dados[1] = dados[1] + 2;
                dados[3] = dados[3] + 1;
            }
            else if (raca == "Elfo")
            {
                dados[2] = dados[2] + 1;
            }
        }

        Console.WriteLine("Ficha preenchida.");


    }

    static void ImprimirFicha(int classe, string nome, string raca, int[] dados, int armor, int vida, int nivel, int xp){
      
      Console.WriteLine("\n--- FICHA DO PERSONAGEM ---");
      Console.WriteLine($"Nome: {nome}");
      Console.Write("Classe: ");
      if (classe == 1)
          Console.WriteLine("Guerreiro");
      else if (classe == 2)
          Console.WriteLine("Mago");
      else if (classe == 3)
          Console.WriteLine("Druida");
      Console.WriteLine($"Raça: {raca}");
      Console.WriteLine("Atributos");
      Console.WriteLine($"Força: {dados[0]}");
      Console.WriteLine($"Destreza: {dados[1]}");
      Console.WriteLine($"Constituição: {dados[2]}");
      Console.WriteLine($"Sabedoria: {dados[3]}");
      Console.WriteLine($"Inteligência: {dados[4]}");
      Console.WriteLine($"Carisma: {dados[5]}");
      Console.WriteLine($"Armor: {armor}");
      Console.WriteLine($"Vida: {vida}");
      Console.WriteLine($"Nível: {nivel}");
      Console.WriteLine($"XP: {xp}");
  }
}
